concepts/src/controllers/pagination.ts

```diff
@@ -0,0 +1,102 @@
+import { URL, URLSearchParams } from "url";
+import { HttpError } from "./error";
+
+const limits = {
+  minSize: 1,
+  maxSize: 100,
+};
+
+const defaultPageSize = 10;
+
+export type PaginationQueryParameters = {
+  page?: number;
+  pageSize?: number;
+};
+
+export type PaginationResponse = {
+  pageSize: number;
+  totalPages: number;
+  totalResults: number;
+  prevPage?: string;
+  nextPage?: string;
+};
+
+export const paginationElasticBody = ({
+  page,
+  pageSize,
+}: PaginationQueryParameters): { from: number; size: number } => {
+  const size = Math.min(pageSize ?? defaultPageSize, limits.maxSize);
```

Comment:
I think this is fine for a hacky POV, but can we get a TODO in here that eventually we'll send users a 400 pointing them to snapshots instead?

Reply:
[We do actually send a 400](https://github.com/wellcomecollection/catalogue-api/blob/0f4f4908e8b59430c8c76419ec84f57c279e8c94/concepts/src/controllers/pagination.ts#L52-L56) with the same message as the prod APIs, but agreed that it should point to the snapshots...